crates/rspack_plugin_javascript/src/visitors/dependency/common_js_scanner.rs

```diff
@@ -106,11 +110,17 @@ impl VisitAstPath for CommonJsScanner<'_> {
     node: &'ast CallExpr,
     ast_path: &mut AstNodePath<AstParentNodeRef<'r>>,
   ) {
-    if is_require_resolve_call(node) {
-      return self.add_require_resolve(node, ast_path, false);
-    }
-    if is_require_resolve_weak_call(node) {
-      return self.add_require_resolve(node, ast_path, true);
+    if let Callee::Expr(expr) = &node.callee {
```

Comment:
You should change this part https://github.com/web-infra-dev/rspack/blob/59ed8c9dc637f19aedf722fef9873bcd713c5d4d/crates/rspack_plugin_javascript/src/visitors/dependency/common_js_import_dependency_scanner.rs#L145 too. We are migrating dependency code generation from AST to string.

Reply:
cc @underfin